Get-DevConsole/InstallDevConsole.psm1

```diff
@@ -0,0 +1,379 @@
+    $currentBuild = Get-IsFirstSetup -currentInstallationDirectory $currentInstallationDirectory
+
+    try
+    {
+        if ($latestBuild -eq $currentBuild.FileVersion)
+        {
+            DisplayCompletedOperation -color Green -text "Update not needed"
+            break
+        }
+        else 
+        {
+            DisplayCompletedOperation -color Green -text "Update needed"
+            $IsFirstSetup = $false
+        }
+    }
+    catch
+    {
+        $IsFirstSetup = $true
+    }   
+    return $IsFirstSetup    
+}
+
+################################################################################
+# Create folder structure to store DevConsole data
+################################################################################
+Function Add-FolderStructure ($directoryName)
+{
+    try 
+    {
+        if (-not (Test-Path -Path $directoryName))
+        {
+            New-Item -ItemType Directory -Path $directoryName > $null
+        }  
+    }
+    catch 
+    {
+        Write-Message -color Red -message "Failed to create DevConsole folder structure"
+        Write-Message -color Red -message "Error: $_"
+    }
+       
+}
+
+################################################################################
+# Terminate DevConsole process
+################################################################################
+Function Stop-DevConsole()
+{
+    try
+    {    
+        Get-process | Where-Object {$_.ProcessName -eq 'DevConsole'} | Stop-Process -Force
+    }
+    catch
+    {
+        Write-Message -color Red -message "Failed to stop active DevConsole session"
+        Write-Message -color Red -message "Error: $_"
+        return $false  
+    }
+}
+
+################################################################################
+# Install latest version of the DevConsole
+################################################################################
+Function Start-InstallDevConsole ($devConsoleBuildFullFileName, $currentInstallationDirectory, $temporaryDownloadDirectory)
+{
+    try
+    {
+        Stop-DevConsole
+        Expand-Archive -Path $("$temporaryDownloadDirectory\$devConsoleBuildFullFileName") -DestinationPath "$currentInstallationDirectory" -Force
+    }
+    catch
+    {
+        Write-Message -color Red -message "Failed to install the DevConsole"
+        Write-Message -color Red -message "Error: $_"
+        retrun $false  
+    }
+
+    return $true
+}
+
+################################################################################
+# Move current installation directory to old cache directory
+################################################################################
+Function Move-CurrentBuildDirectory($isFirstSetup)
+{
+    try
+    {
+        #Move current build to old cache
+        if ($false -eq $isFirstSetup)
+        {
+            $Date = (Get-Date -Format d).Replace("/","-")
+            $currentCache = Get-ChildItem -Path "$currentInstallationDirectory"

```

Comment:
$currentInsallationDirectory does not exist within this scope. I also see "oldCacheDirectory", below. You should pass these variables in or make them global. I know you declare them in a function that calls this function. I know that languages like C# will not scope variables in a function to function calls (not sure if PS does or not), but even if Powershell did, you shouldn't assume the variables are inside the scope of the calling function. What if the script changes in a year?

Reply:
I added parameters to the function to address this concern.